UNITTESTS/fakes/drivers/watchdog/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-os-fakes-drivers-watchdog)
+
+target_compile_definitions(mbed-os-fakes-drivers-watchdog 
+    PUBLIC
+        DEVICE_WATCHDOG
+        MBED_WDOG_ASSERT=1
+)
+
+target_sources(mbed-os-fakes-drivers-watchdog
+    PRIVATE
+        ${MBED_PATH}/drivers/source/Watchdog.cpp
+)
+
+target_link_libraries(mbed-os-fakes-drivers-watchdog
+    PUBLIC
```

Comment:
shouldn't these (and in other libs) we PRIVATE? otherwise it will interfere with someone using these in their test when they don't want to include all the headers

Reply:
This watchdog fake library is consumable by test watchdog unittest and did not have any header conflicts.
If fakes library make any header conflicts then we go with PRIVATE

Reply:
What's the benefit of having it PUBLIC?

Reply:
If they are PUBLIC then if you depend on this library you also depend on the libraries it depends on.

This is like a header relying on another header to include something rather than including it itself. I think it's better if things are self contained and just exposing thing they are meant to expose and not what they are using themselves.

Reply:
The hal watchdog.h which is part of mbed-os-headers interface library, is required by both test_watchdog unit test and fake watchdog library.

The fake watchdog library has target_link_library of mbed-os-headers with PUBLIC so that watchdog.h will be available to test_watchdog by just target link to the fake library instead of reincluding mbed-os-headers in test_watchdog CMakeLists.txt.

I couldn't find any advantage to make fakes watchdog CMake target link libraries of mbed-os-headers to PRIVATE and reinclude headers lib in test_watchdog CMake :)

Reply:
I do because if I rely on your fake watchdog but don't want to include the mbed-os-headers because I need some fake header I'm screwed.